pkg/volume/azure_dd/azure_common_linux.go

```diff
@@ -77,6 +77,19 @@ func findDiskByLunWithConstraint(lun int, io ioHandler, azureDisks []string) (st
 			if len(arr) < 4 {
 				continue
 			}
+			if len(azureDisks) == 0 {
+				// there is nothing under /dev/disk/azure/
+				target, err := strconv.Atoi(arr[0])
+				if err != nil {
+					glog.Errorf("failed to parse target from %v (%v), err %v", arr[0], name, err)
+					continue
+				}
+				// as observed, targets 0-3 are used by OS disks. Skip them
```

Comment:
I used to use this heuristics but it is not reliable: on certain instances (last time I used F instance and found this test was different than D instance), this test fails.

Reply:
which test are you refering to? You mean ""/sys/bus/scsi/devices"" on different VM instance is different?

Reply:
hi @rootfs I have tried coreos on `Standard F1s ` VM, it works well. Do you have any other instance suggestion?

Reply:
Paste all the info and logs on F instance, I attached two disks and try to detect Lun 1 disk:
```
azureuser@andy-coreosf /tmp $ ./detect-new-lun1
Azure sys disks paths: []there is nothing under /dev/disk/azure/
there is nothing under /dev/disk/azure/
there is nothing under /dev/disk/azure/
there is nothing under /dev/disk/azure/
/dev/sdd <nil>
azureuser@andy-coreosf /tmp $ ls /sys/bus/scsi/devices -lt
total 0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 target0:0:0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0000-8899-0000-000000000000/host0/target0:0:0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 target3:0:1 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0001-8899-0000-000000000000/host3/target3:0:1
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 target5:0:0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/f8b3781b-1e82-4818-a1c3-63d806ec15bb/host5/target5:0:0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 5:0:0:0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/f8b3781b-1e82-4818-a1c3-63d806ec15bb/host5/target5:0:0/5:0:0:0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 5:0:0:1 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/f8b3781b-1e82-4818-a1c3-63d806ec15bb/host5/target5:0:0/5:0:0:1
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host5 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/f8b3781b-1e82-4818-a1c3-63d806ec15bb/host5
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 3:0:1:0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0001-8899-0000-000000000000/host3/target3:0:1/3:0:1:0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host3 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0001-8899-0000-000000000000/host3
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host4 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/f8b3781a-1e82-4818-a1c3-63d806ec15bb/host4
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 0:0:0:0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0000-8899-0000-000000000000/host0/target0:0:0/0:0:0:0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host0 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A03:00/device:07/VMBUS:01/00000000-0000-8899-0000-000000000000/host0
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host1 -> ../../../devices/pci0000:00/0000:00:07.1/ata1/host1
lrwxrwxrwx. 1 root root 0 Oct 23 02:52 host2 -> ../../../devices/pci0000:00/0000:00:07.1/ata2/host2
```

Reply:
`as observed, targets 0-3 are used by OS disks`, there are similar observations [too](https://github.com/Azure/azure-sdk-for-go/issues/315#issuecomment-215212498)

However, this observation failed when i was using an F8s instance running RHEL. And that gave rise to this [fix](https://github.com/kubernetes/kubernetes/pull/44295)

Reply:
Hi @rootfs , thanks for the check. 
The latest code already contains https://github.com/kubernetes/kubernetes/pull/44295, while current coreos issue is: since `/dev/disk/azure/` is not populated in coreos, which means there would be no filtering, see: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/azure_dd/azure_common_linux.go#L119-L124. In the end, `findDiskByLun` would return a wrong dev name which is actually os disk. So my fix is add back the skip logic if `azureDisks` is empty.
I have also tested on RHEL F instance, my PR works well. Let me know if there is any other issue, thx.

Reply:
@andyzhangx thanks, please glog a message here.

Reply:
@rootfs done